app/models/order.rb

```diff
@@ -1,6 +1,8 @@
 # frozen_string_literal: true
 
 class Order < ApplicationRecord
+  serialize :snapshot, Reward
```

Comment:
nit: the name might be a bit ambiguous. I would call it `reward_snapshot`

Reply:
good point :)